server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -115,11 +129,21 @@ public void prepareAndValidateBackup() {
             String zipPath = cacheRoot.resolve(configFile).toString();
             try (InputStream is = zip.getInputStream(zip.getEntry(zipPath))) {
                ConfigurationBuilderHolder builderHolder = parserRegistry.parse(is, null, MediaType.fromExtension(configFile));
-               Configuration cfg = builderHolder.getNamedConfigurationBuilders().get(cacheName).build();
-               log.debugf("Restoring Cache %s: %s", cacheName, cfg.toXMLString(cacheName));
-               // GetOrCreate in the event that a default cache is defined. This also allows cache-configurations to be
-               // modified prior to a restore if only the backed up data is required
-               SecurityActions.getOrCreateCache(cm, cacheName, cfg);
+               Configuration config = builderHolder.getNamedConfigurationBuilders().get(cacheName).build();
+               log.debugf("Restoring Cache %s: %s", cacheName, config.toXMLString(cacheName));
+               Map<String, String> params = new HashMap<>(2);
+               params.put("name", cacheName);
+               params.put("configuration", config.toXMLString(cacheName));
+               TaskContext taskContext = new TaskContext();
+               taskContext.cacheManager(cm);
+               taskContext.parameters(params);
+
+               CompletionStage<?> taskExecution = SecurityActions.executeTask(BackupCreateCacheTask.NAME, taskManager, taskContext);
```

Comment:
I avoided using the cluster executor directly so that I didn't have to handle the marshalling of the cache name & configuration parameters. Tasks handle this already via `DistributedServerTask` and `TaskParameter`.